framework.py

```diff
@@ -9,32 +9,166 @@
 # not expressly granted therein are reserved by Shotgun Software Inc.
 
 import sgtk
+import sys
 import os
+import struct
+from sgtk.util import LocalFileStorageManager
+
 
 class DesktopserverFramework(sgtk.platform.Framework):
+    """
+    Provides browser integration.
+    """
+
+    def __init__(self, *args, **kwargs):
+        super(DesktopserverFramework, self).__init__(*args, **kwargs)
+        self._server = None
+        self._settings = None
+        self._tk_framework_desktopserver = None
 
     ##########################################################################################
     # init and destroy
+    def launch_desktop_server(self):
```

Comment:
There are two reasons for doing this.
Frameworks are always initialized before the engine that uses them. This cause the styling of the widget to be the boring Qt default instead of the sleek Desktop one. So this is getting called after the desktop has set its look and feel.

Second, you have to remember that this is the first step of the desktop server integration into the engine. In the second step, the desktop will be aware of the server because it will need to install callbacks to restart the app or reauthenticate when requests are coming from another site.